product/scanning.py

```diff
@@ -94,6 +94,14 @@ def retrieve_scan_result(
         product = self._api.scan_result(product_id=product_id)
         return product
 
+    @lru_cache(maxsize=8)
+    def _retrieve_cached_scan_result(
+        self,
+        container_image: ContainerImage,
+        component: Component,
+    ):
+        return self.retrieve_scan_result(container_image=container_image, component=component)
```

Comment:
caching may make sense. but why introduce it now? also: why not always cache (as in: why a new function)?

Reply:
Caching the scan results allows me to retrieve them for free exactly when I need them instead of passing them to every function if they _might_ be needed.